### PR TITLE
Implementation of IGG to 2D stokes solvers

### PR DESCRIPTION
**Pull request by aelligp**
- implements Implicit global grid to 2D solvers

- updates test and benchmarks

### CODE REVIEW

File: miniapps/benchmarks/stokes/solvi/SolVi.jl [hunk 1]
```diff
@@ -48,18 +48,23 @@ function solvi_viscosity(ni, di, li, rc, η0, ηi)
...
-    di = @. li / ni # grid step in x- and -y
-    max_li = max(li...)
-    nDim = length(ni) # domain dimension
-    xci = Tuple([(di[i] / 2):di[i]:(li[i] - di[i] / 2) for i in 1:nDim]) # nodes at the center of the cells
-    xvi = Tuple([0:di[i]:li[i] for i in 1:nDim]) # nodes at the vertices of the cells
+    origin = zero(nx), zero(ny)
+    igg = IGG(init_global_grid(nx, ny, 1; init_MPI=init_MPI)...) # init MPI
+    di = @. li / (nx_g(), ny_g()) # grid step in x- and -y
+    xci, xvi = lazy_grid(di, li, ni; origin=origin) # nodes at the center and vertices of the cells
+
```
Comment (albert-de-montserrat) [inline miniapps/benchmarks/stokes/solvi/SolVi.jl:62]: remove thse comented lines

File: miniapps/benchmarks/stokes/solvi/SolViEl.jl [hunk 2]
```diff
@@ -38,17 +38,23 @@ function solvi_viscosity(xci, ni, li, rc, η0, ηi)
...
     li = (lx, ly)  # domain length in x- and y-
-    di = @. li / ni # grid step in x- and -y
-    nDim = length(ni) # domain dimension
-    xci = Tuple([(di[i] / 2):di[i]:(li[i] - di[i] / 2) for i in 1:nDim]) # nodes at the center of the cells
-    xvi = Tuple([0:di[i]:li[i] for i in 1:nDim]) # nodes at the vertices of the cells
+    origin = zero(nx), zero(ny)
+    igg = IGG(init_global_grid(nx, ny, 1; init_MPI=init_MPI)...) #init MPI
+    di = @. li / (nx_g(), ny_g()) # grid step in x- and -y
+    xci, xvi = lazy_grid(di, li, ni; origin=origin) # nodes at the center and vertices of the cells
+
```
Comment (albert-de-montserrat) [inline miniapps/benchmarks/stokes/solvi/SolViEl.jl:52]: remove these comented lines

**Review by albert-de-montserrat — CHANGES_REQUESTED** [review]

**Comment by aelligp** [comment]
3D benchmarks need revision due to boundary conditions - otherwise IGG implementation is done

**Review by albert-de-montserrat — COMMENTED** [review]
I think that [here](https://github.com/PTsolvers/JustRelax.jl/blob/ad19be269569a941ce925d4c2ba09f9bd22f6546/miniapps/benchmarks/stokes/solcx/SolCx.jl#L16) - and probably in some of the other benchmarks- you need to create the coordinate array using `x_g` / ''y_g' from [`ImplicitGlobalGrid.jl`](https://github.com/eth-cscs/ImplicitGlobalGrid.jl). Otherwise, it will define the wrong geometry when using multiple processes.

**Pull request merged**